test/Window/WindowBase.test.cpp

```diff
@@ -104,14 +105,38 @@ TEST_CASE("[Window] sf::WindowBase", runDisplayTests())
 
     SECTION("pollEvent()")
     {
-        sf::WindowBase windowBase;
-        CHECK(!windowBase.pollEvent());
+        SECTION("Uninitialized window")
+        {
+            sf::WindowBase windowBase;
+            CHECK(!windowBase.pollEvent());
+        }
     }
 
     SECTION("waitEvent()")
     {
-        sf::WindowBase windowBase;
-        CHECK(!windowBase.waitEvent());
+        SECTION("Uninitialized window")
+        {
+            sf::WindowBase windowBase;
+            CHECK(!windowBase.waitEvent());
+        }
+
+        SECTION("Initialized window")
+        {
+            sf::WindowBase windowBase(sf::VideoMode({360, 240}), "WindowBase Tests");
+
+            constexpr auto timeout = sf::milliseconds(250);
+
+            const auto startTime = std::chrono::steady_clock::now();
+            const auto event     = windowBase.waitEvent(timeout);
+            const auto elapsed   = std::chrono::steady_clock::now() - startTime;
```

Comment:
I recommend using an `sf::Clock`

Reply:
I would recommend against it for the same reasons described in https://github.com/SFML/SFML/pull/3094#discussion_r1635519805. Resolving right now, feel free to re-open if you feel strongly about this recommendation.